samples/basic/threads/src/main.c

```diff
@@ -41,18 +25,40 @@ struct printk_data_t {
 
 K_FIFO_DEFINE(printk_fifo);
 
-void blink(const char *port, u32_t sleep_ms, u32_t led, u32_t id)
+#ifndef DT_ALIAS_LED0_GPIOS_FLAGS
+#define DT_ALIAS_LED0_GPIOS_FLAGS 0
+#endif
+
+#ifndef DT_ALIAS_LED1_GPIOS_FLAGS
+#define DT_ALIAS_LED1_GPIOS_FLAGS 0
+#endif
+
+struct led {
+	const char *gpio_dev_name;
+	const char *gpio_pin_name;
+	unsigned int gpio_pin;
+	unsigned int gpio_flags;
+};
+
+void blink(const struct led *led, u32_t sleep_ms, u32_t id)
 {
-	int cnt = 0;
 	struct device *gpio_dev;
+	int cnt = 0;
+	int ret;
 
-	gpio_dev = device_get_binding(port);
-	__ASSERT_NO_MSG(gpio_dev != NULL);
+	gpio_dev = device_get_binding(led->gpio_dev_name);
+	__ASSERT(gpio_dev != NULL, "Error: didn't find %s device\n",
+			led->gpio_dev_name);
```

Comment:
Would be more consistent sing printk, similarly as what is done below

Reply:
The blinky and button samples use `printk` in this place so this is indeed not consistent. However, the ASSERT version is closer to what the real application would look like. Checking the result of `device_get_binding` is only required during the development phase. Once we provide a correct device name the check can be turned off. On the other hand the call to `gpio_pin_configure` can fail during the run time if we are working with an external GPIO driver.

I would rather stay with the existing approach.

Reply:
> The blinky and button samples use printk in this place so this is indeed not consistent. However, the ASSERT version is closer to what the real application would look like. Checking the result of device_get_binding is only required during the development phase. Once we provide a correct device name the check can be turned off.

This makes sense, but from what I see this isn't a widely used approach accros samples.
I'm approving the change, but it would be interesting to have this proposed (and potentially discussed) as a general guidelines for samples implementation.